Warnings generated where member functions are not marked as 'override'.

**Describe the bug**
Compiling GQCP (develop branch) generates a whole lot of warnings `/Users/dariatolstykh/work/gqcg/GQCP/gqcp/include/Basis/SpinorBasis/SimpleSpinorBasis.hpp:162:22: warning: 'transformed' overrides a member function but is not marked 'override'` during compilation. 

**To Reproduce**
Compiling develop without warnings.

**Expected behavior**
Compilation without warnings.

**Screenshots and logs**
![Screenshot 2020-10-30 at 09 25 48](https://user-images.githubusercontent.com/33431368/97677048-ef29b480-1a91-11eb-934c-39016e846e51.png)

**Operating system**
 - OS: macOS
 - Version 10.15.7

--- a/gqcp/include/Basis/SpinorBasis/SimpleSpinorBasis.hpp
+++ b/gqcp/include/Basis/SpinorBasis/SimpleSpinorBasis.hpp
@@ -159,7 +159,7 @@ public:
      * 
      *  @return The basis-transformed object.
      */
-    FinalSpinorBasis transformed(const TM& transformation_matrix) const {
+    FinalSpinorBasis transformed(const TM& transformation_matrix) const override {
 
         auto result = this->derived();
         result.C.transform(transformation_matrix);

